components/core/src/fs.rs

```diff
@@ -694,16 +709,34 @@ fn parent(p: &Path) -> io::Result<&Path> {
 ///
 /// Assumes that the parent directory of dest_path exists.
 pub struct AtomicWriter {
-    dest:     PathBuf,
-    tempfile: tempfile::NamedTempFile,
+    dest:       PathBuf,
+    tempfile:   tempfile::NamedTempFile,
+    #[cfg(not(windows))]
+    permission: Option<u32>,
+    #[cfg(windows)]
+    permission: Option<&[win_perm::PermissionEntry]>,
```

Comment:
This doesn't actually work, it was just my first swing at what we might want to do, but for this to work I think we'll need some lifetime annotations.  If a reviewer things this is a useful feature for windows, I'll put some time into getting a cross-compilation setup going so I can develop the window's portions, otherwise, I'll make this PR posix-only for now since it is posix-only at the call-site.

Reply:
I think a Vec<PermissionEntry> is probably the way to go here if we proceed with the windows stuff, thanks.